lib/cocoapods/command/repo/push.rb

```diff
@@ -29,6 +32,7 @@ def initialize(argv)
           @allow_warnings = argv.flag?('allow-warnings')
           @local_only = argv.flag?('local-only')
           @repo = argv.shift_argument
+          @source_urls = argv.option('sources', SourcesManager.all.map(&:url).join(',')).split(',')
```

Comment:
I think its nice if you specify the sources option not to add anything magically. Up to you though! I can change it. I verified the change works in our `repo push` command for our internal pod.